packages/cli/src/lingui-compile.ts

```diff
@@ -35,23 +44,25 @@ function command(config: LinguiConfig, options) {
   const doMerge = !!config.catalogsMergePath
   let mergedCatalogs = {}
 
-  console.error("Compiling message catalogs…")
+  console.log("Compiling message catalogs…")
 
-  config.locales.forEach((locale) => {
+  for (const locale of config.locales) {
     const [language] = locale.split(/[_-]/)
+    // todo: this validation should be in @lingui/conf
     if (locale !== config.pseudoLocale && !plurals[language]) {
-      console.log(
+      console.error(
         chalk.red(
           `Error: Invalid locale ${chalk.bold(locale)} (missing plural rules)!`
         )
       )
       console.error()
+      // todo: should we exit here?
```

Comment:
As written in the todo this validation should be in `@lingui/conf`, i checked the git history and looks like it's here by historical reasons (all validations were here at the beginning) 

I didn't move that because it's out of the scope of current PR, and you don't like big PRs, right 😉?

Reply:
If possible, it's always a good idea to keep PRs small or at least create separate commits for unrelated changes 🙂 

We can keep this validation here for a while. As I understand, if it were in `@lingui/conf`, it would prevent such an error at an earlier stage?

Regarding the second `todo` - should we commit such todos? It risks remaining eternal todo

Reply:
> Regarding the second todo - should we commit such todos? It risks remaining eternal todo
It's a question to maintainers here. What do you think, should we exit here? 

Pros: 

- If you run this in CI, it will return `1` and will fail job and you got your error earlier. 

Cons: 

I don't know how users use it and how can it affect them. Maybe they use some not-real locale intentionally?

Reply:
It could be some custom language

Reply:
> We can keep this validation here for a while. As I understand, if it were in @lingui/conf, it would prevent such an error at an earlier stage?

It's ideologically should be in @lingui/conf. All other validation for locales applied there.
Moreover, I think the most popular setup now is not using `cli compile` at all, instead it uses loaders (webpack, etc) and this check would never be applied. 

Also, this check could be useful in other places, not only in compile.

Reply:
> It could be some custom language

For what? If it is a pseudolocale it would not throw error. So i think in the vast majority of cases it would be user misconfiguration.

Reply:
That makes sense

Reply:
> should we exit here?

Anyway, when it would be moved to @lingui/config it would not be possible to early return (at least with current implementation)

Reply:
It would be indeed better to exit with the `1` exit code here. It's one of the common practices for CLI tools - exiting with this code in case of errors.

Reply:
@thekip @andrii-bodnar There was a `process.exit(1)` here in the past. But as PR #1280 (which removed it) correctly pointed out, there are some languages which don't have plurals ([are not present in CLDR](https://unicode-org.github.io/cldr-staging/charts/latest/supplemental/language_plural_rules.html)). So I think we don't think we should exit here at all.

Reply:
@Martin005 nice catch. I think detecting validity of locale by checking the absence of plural-rules is incorrect in general. I think we still need validate them and exit if error, but use cldr for that.  This is of course out of the context of this PR. I will remove the `todo` here